### PR TITLE
Added Configuration.WithExpandExceptionLogs

### PR DESCRIPTION
**Pull request by Falco20019**
Right now, when using `Configuration`, it's not possible to call `WithExpandExceptionLogs()` on the `TraceBuilder` without using `GetTracerBuilder()` and Build it directly instead of using `GetTracer()`. It seems, that this is nearly the only flag that can not be set by environment.

I assume this should be discussed and proposed in the meta repo? But I first wanted to hear if @cwe1ss or @yurishkuro already have an opionion on that.

### CODE REVIEW

**Comment by yurishkuro** [comment]
is there a situation where we'd want to *not* capture the stacktrace? Maybe it should be the default behavior without requiring additional configuration.

**Comment by Falco20019** [comment]
I assume this is mostly a compromise between having redundant data to search tags or only have all the information not searchable in error.object

How is this handled in Java?

**Comment by yurishkuro** [comment]
we don't have a well defined process for this at the moment in any client library. I was not thinking of making the stack traces searchable, but just to capture the stacktrace fully. Other tracing formats (e.g. [x-ray][1] and [census][2]) have a well defined structure in the span for capturing exceptions. We may want to extend the upcoming protobuf format with that as well.

[1]: https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-errors
[2]: https://github.com/census-instrumentation/opencensus-proto/blob/master/opencensus/proto/trace/trace.proto#L277

**Comment by Falco20019** [comment]
@yurishkuro Just checked the [java implementation](https://github.com/jaegertracing/jaeger-client-java/blob/2fa8f3ed17e88114f6cd7114747ed298933dabd8/jaeger-core/src/main/java/io/jaegertracing/Span.java#L273) and it's the same there. Default seems to be using `error.object` and copy the full exception/throwable and if `WithExpandExceptionLogs()` is called on the `Tracer.Builder`, it's decomposed into `error.kind`, `message` and `stack`. The stack is never decomposed any further, but I think that's fine for now.

So even in Java, it's currently not possible to set `WithExpandExceptionLogs` directly on `Configuration` without having to get the `Tracer.Builder` and call it on that instead of getting the `Tracer` directly from `Configuration`.

**Comment by Falco20019** [comment]
Maybe we could add this to v0.2.0 since we are already want to make changes to Configuration.

**Comment by Falco20019** [comment]
@yurishkuro This is the only configuration flag right now that is not settable through environment. We added `JAEGER_TRACEID_128BIT` yesterday, so maybe we could also add `JAEGER_EXPAND_EXCEPTION_LOGS`. At least `Java` also has the `WithExpandExceptionLogs` so it's not just a C# feature. As written above, default behaviour is to not expand to reduce redundant logs for the costs of it being harder filterable. Maybe also relates to https://github.com/jaegertracing/documentation/issues/105

**Comment by yurishkuro** [comment]
what would be a situation when someone would not want to expand exceptions this way?

**Comment by Falco20019** [comment]
Already part of the Tracer.Builder API. So both ways are supported then

**Comment by yurishkuro** [comment]
I realize that, my question is on the broader scope of this being a standard env var across languages, do we have enough justification for this behavior to be configurable rather than "always expand"?

**Comment by Falco20019** [comment]
I assume it’s a legitimate decision to choose between non-redundant storage (not expanded) and searchability (expanded) and a decision the developer has to make.

**Comment by yurishkuro** [comment]
If you look at opencensus (and probably x-ray) span data models, they have explicit data structure for exceptions and the stack trace.

> I assume it’s a legitimate decision to choose between non-redundant storage (not expanded) and searchability (expanded) 

Is it fair to rephrase this as a trade-off between expressiveness of the captured data and overhead of collecting & storing it (I'd argue mainly collecting since it's a client library). In this case, what concerns me with an explicit option for just a single aspect that affects this trade-off is where it ends. For example, I remember Lightstep tracer has an option on the max number of logs to keep in a span. Does the user actually care?  If the only concern here is the performance trade-off, as a user I might simply want a single knob that I can turn from 0 to 1 for min overhead to max expressiveness. In other words, the user defines a certain overhead threshold, and it's up to the tracer to decide how it achieves that. Maybe one span has 100 logs and so there's no space for the stacktrace, while the other has a single log and a stracktrace would be perfectly fine to capture.

**Comment by yurishkuro** [comment]
> Is this already solved in any of the other libraries?

no

**Comment by Falco20019** [comment]
I will keep that for now that way since we should at least have the full API configurable. But I would add the discussion above into a separate issue for the future. I think you have a point that the user should not have to care about low-level flags but about behaviour. This will need some adjustments to the Tracer and maybe some other classes, which is definetly not the focus of this `Configuration` enhancement.

File: src/Jaeger/Configuration.cs [hunk 1]
```diff
@@ -272,6 +284,12 @@ public Configuration WithTracerTags(Dictionary<string, string> tracerTags)
             return this;
         }
 
+        public Configuration WithExpandExceptionLogs(bool expandExceptionLogs=true)
```
Comment (Falco20019) [inline src/Jaeger/Configuration.cs]: Not sure if we should do it using the default value. The idea was to keep it similar to [Tracer.Builder.WithExpandExceptionLogs()](https://github.com/jaegertracing/jaeger-client-csharp/blob/68dac3108ed4c020245a58ce484ba9722ec6981e/src/Jaeger/Tracer.cs#L295), but also have it resettable.

**Review by yurishkuro — APPROVED** [review]
